src/backend/web.rs

```diff
@@ -0,0 +1,464 @@
+    mapped.depth_fail_op(map_stencil_operation(desc.depth_fail_op));
+    mapped.fail_op(map_stencil_operation(desc.fail_op));
+    mapped.pass_op(map_stencil_operation(desc.pass_op));
+    mapped
+}
+
+fn map_depth_stencil_state_descriptor(desc: &wgt::DepthStencilStateDescriptor) -> web_sys::GpuDepthStencilStateDescriptor {
+    let mut mapped = web_sys::GpuDepthStencilStateDescriptor::new(map_texture_format(desc.format));
+    mapped.depth_compare(map_compare_function(desc.depth_compare));
+    mapped.depth_write_enabled(desc.depth_write_enabled);
+    mapped.stencil_back(&map_stencil_state_face_descriptor(&desc.stencil_back));
+    mapped.stencil_front(&map_stencil_state_face_descriptor(&desc.stencil_front));
+    mapped.stencil_read_mask(desc.stencil_read_mask);
+    mapped.stencil_write_mask(desc.stencil_write_mask);
+    mapped
+}
+
+fn map_blend_descriptor(desc: &wgt::BlendDescriptor) -> web_sys::GpuBlendDescriptor {
+    let mut mapped = web_sys::GpuBlendDescriptor::new();
+    mapped.dst_factor(map_blend_factor(desc.dst_factor));
+    mapped.operation(map_blend_operation(desc.operation));
+    mapped.src_factor(map_blend_factor(desc.src_factor));
+    mapped
+}
+
+fn map_blend_factor(factor: wgt::BlendFactor) -> web_sys::GpuBlendFactor {
+    use wgt::BlendFactor;
+    use web_sys::GpuBlendFactor as bf;
+    match factor {
+        BlendFactor::Zero => bf::Zero,
+        BlendFactor::One => bf::One,
+        BlendFactor::SrcColor => bf::SrcColor,
+        BlendFactor::OneMinusSrcColor => bf::OneMinusSrcColor,
+        BlendFactor::SrcAlpha => bf::SrcAlpha,
+        BlendFactor::OneMinusSrcAlpha => bf::OneMinusSrcAlpha,
+        BlendFactor::DstColor => bf::DstColor,
+        BlendFactor::OneMinusDstColor => bf::OneMinusDstColor,
+        BlendFactor::DstAlpha => bf::DstAlpha,
+        BlendFactor::OneMinusDstAlpha => bf::OneMinusDstAlpha,
+        BlendFactor::SrcAlphaSaturated => bf::SrcAlphaSaturated,
+        BlendFactor::BlendColor => bf::BlendColor,
+        BlendFactor::OneMinusBlendColor => bf::OneMinusBlendColor,
+    }
+}
+
+fn map_blend_operation(op: wgt::BlendOperation) -> web_sys::GpuBlendOperation {
+    use wgt::BlendOperation;
+    use web_sys::GpuBlendOperation as bo;
+    match op {
+        BlendOperation::Add => bo::Add,
+        BlendOperation::Subtract => bo::Subtract,
+        BlendOperation::ReverseSubtract => bo::ReverseSubtract,
+        BlendOperation::Min => bo::Min,
+        BlendOperation::Max => bo::Max,
+    }
+}
+
+fn map_index_format(format: wgt::IndexFormat) -> web_sys::GpuIndexFormat {
+    use wgt::IndexFormat;
+    use web_sys::GpuIndexFormat as f;
+    match format {
+        IndexFormat::Uint16 => f::Uint16,
+        IndexFormat::Uint32 => f::Uint32,
+    }
+}
+
+fn map_vertex_format(format: wgt::VertexFormat) -> web_sys::GpuVertexFormat {
+    use wgt::VertexFormat;
+    use web_sys::GpuVertexFormat as vf;
+    match format {
+        VertexFormat::Uchar2 => vf::Uchar2,
```

Comment:
I wonder if these maps are truly necessary. Perhaps, we can always say that our enum values match the target WebIDL, and do a transmute?

Reply:
These are strings in WebIDL, see https://docs.rs/web-sys/0.3.36/src/web_sys/features/gen_GpuVertexFormat.rs.html#4